source/uwp/Renderer/lib/MediaHelpers.cpp

```diff
@@ -179,24 +179,97 @@ namespace AdaptiveCards::Rendering::Uwp::MediaHelpers
+                {
+                    if (currentCaptionType == supportedCaptionType)
+                    {
+                        const auto timedTextURL = GetUrlFromString(renderContext.HostConfig(), captionSource.Url());
+
+                        winrt::IAdaptiveCardResourceResolver resourceResolver{nullptr};
+                        if (const auto resourceResolvers = renderContext.ResourceResolvers())
+                        {
+                            resourceResolver = resourceResolvers.Get(timedTextURL.SchemeName());
+                        }
+
+                        const auto timedTextSrcResolvedHelper = [label = captionSource.Label()](winrt::TimedTextSource const& /*sender*/,
+                                                                                                winrt::TimedTextSourceResolveResultEventArgs const& args)
+                                {
+                                    if (!args.Error())
+                                    {
+                                        args.Tracks().GetAt(0).Label(label);
+                                    }
+                                };
+                        if (resourceResolver == nullptr)
+                        {
+                            const auto timedTextSrc = winrt::TimedTextSource::CreateFromUri(timedTextURL);
+                            timedTextSrc.Resolved(timedTextSrcResolvedHelper);
+                            mediaSrc.ExternalTimedTextSources().Append(timedTextSrc);
+                        }
+                        else
+                        {
+                            auto args = winrt::make<winrt::implementation::AdaptiveCardGetResourceStreamArgs>(timedTextURL);
+                            auto getResourceStreamOperation = resourceResolver.GetResourceStreamAsync(args);
+                            getResourceStreamOperation.Completed(
+                                [mediaSrc, timedTextSrcResolvedHelper](winrt::IAsyncOperation<winrt::IRandomAccessStream> operation,
+                                                                                       winrt::AsyncStatus status) -> void
+                                {
+                                    if (status == winrt::AsyncStatus::Completed)
+                                    {
+                                        // Get the random access stream
+                                        if (const auto randomAccessStream = operation.GetResults())
+                                        {
+                                            auto timedTextSrc = winrt::TimedTextSource::CreateFromStream(randomAccessStream);
+                                            timedTextSrc.Resolved(timedTextSrcResolvedHelper);
+                                            mediaSrc.ExternalTimedTextSources().Append(timedTextSrc);
+                                        }
+                                    }
+                                });
```

Comment:
I tried this, (calling `randomAccessStream.get()` since `randomAccessStream` right now is just an `IAsyncOperation`), and I get this compiler error:

`'promise_type': is not a member of 'std::experimental::coroutine_traits<void,const winrt::Windows::UI::Xaml::Controls::MediaElement &,const winrt::AdaptiveCards::ObjectModel::Uwp::AdaptiveMedia &,const winrt::AdaptiveCards::Rendering::Uwp::AdaptiveRenderContext &,const winrt::Windows::Media::Core::MediaSource &>'`

Reply:
I tried adding `#include <winrt/Windows.Foundation.h>` to `CppWinRTIncludes.h` (guidance from [this article](https://devblogs.microsoft.com/oldnewthing/20210809-00/?p=105539)) and that did not fix it

Reply:
@paulcam205 we can't use `co_await` since we're not inside a co routine. We can use a blocking `.get()`.
@PankajBhojwani I've just tried and `.get()` call compiles fine without any issue:)

Reply:
Ah yes removing the `co_await` and adding `get` works, thanks!